pallets/settlement/src/lib.rs

```diff
@@ -870,6 +881,12 @@ impl<T: Config> Module<T> {
         value_date: Option<T::Moment>,
         legs: Vec<Leg>,
     ) -> Result<u64, DispatchError> {
+        // Ensure instruction does not have too many legs.
+        ensure!(
+            (legs.len() as u32) <= T::MaxLegsInInstruction::get(),
```

Comment:
```suggestion
            legs.len() <= T::MaxLegsInInstruction::get() as u32,
```
Otherwise you can truncate the length of the vector. The suggested change is more restrictive (good).

Reply:
Shouldn't the suggested code be: ` as usize` instead of ` as u32`?

Reply:
```suggestion
            legs.len() <= T::MaxLegsInInstruction::get() as usize,
```